resources/lib/controller.py

```diff
@@ -499,12 +499,19 @@ def start_playback():
 
     utils.crunchy_log("Starting loop", xbmc.LOGINFO)
     # stay in this method while playing to not lose video_player, as backgrounds threads reference it
-    while video_player.is_playing():
-        time.sleep(1)
```

Comment:
The problem can be solved in the following simple way:
But it's not a nice solution and I had some strange things happen when I played the next video. That's why I removed the threading.
[https://xbmc.github.io/docs.kodi.tv/master/kodi-base/d6/db2/group__python__xbmc.html#gac81db027eab4f3b5bec4a7a5a1073e7e](https://xbmc.github.io/docs.kodi.tv/master/kodi-base/d6/db2/group__python__xbmc.html#gac81db027eab4f3b5bec4a7a5a1073e7e)
```suggestion
    # stay in this method while playing to not lose video_player, as backgrounds threads reference it
    xbmc.sleep(2000) #wait 2 sec
    while video_player.is_playing():
        time.sleep(1)
```
As Info:
You should use monitor.abortRequested() and monitor.waitForAbort(1) in loops so that Kodi can terminate the loops normally and the addon does not have to be killed in the event of an abort.
[https://xbmc.github.io/docs.kodi.tv/master/kodi-base/de/d93/group__python__monitor.html#ga9a1d271d0fddebee906abcda904e8249](https://xbmc.github.io/docs.kodi.tv/master/kodi-base/de/d93/group__python__monitor.html#ga9a1d271d0fddebee906abcda904e8249)
```suggestion
    # stay in this method while playing to not lose video_player, as backgrounds threads reference it
    xbmc.sleep(2000) #wait 2 sec
    while not xbmc.monitor.abortRequested() and video_player.is_playing():
       xbmc.monitor.waitForAbort(1) 
```